runelite-client/src/main/java/net/runelite/client/plugins/prayerindicator/PrayerIndicatorOverlay.java

```diff
@@ -0,0 +1,269 @@
+	
+	private Color orbIndicatorColor = new Color (0, 255, 255, 255);
+	private int alphaOffset = -10;			// How much to subtract/add to the alpha of the indicator orb
+	private int playerPotionID;				// The current ItemID of the potion the player is carrying
+	private BufferedImage prayPotIcon;
+	private Counter indicator;				// The indicator that will go in the info box if the player chooses to have it shown
+	
+	//private PrayerIndicatorPlugin plugin;
+	private final Client client;
+	private final PrayerIndicatorConfig config;
+	private final InfoBoxManager infoBoxManager;
+	private final QueryRunner queryRunner;
+	
+	@Inject
+	private ItemManager itemManager;
+	
+	@Inject
+	private PrayerIndicatorPlugin plugin;
+	
+	@Inject
+	public PrayerIndicatorOverlay (QueryRunner queryRunner, Client client, PrayerIndicatorConfig config, InfoBoxManager infoBoxManager)
+	{
+		setPosition (OverlayPosition.DYNAMIC);
+		setLayer (OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.infoBoxManager = infoBoxManager;
+		this.queryRunner = queryRunner;
+	}
+	
+	// This method will be called every game tick and will be used to fade the orb indicator in and out
+	public void onTick ()
+	{
+		//Every game tick let's get what kind of prayer potion the player has
+		playerPotionID = getPlayerPotion ();
+		
+		//Let's make sure the infoBox indicator is set
+		if (indicator == null)
+		{
+			prayPotIcon = itemManager.getImage (ItemID.PRAYER_POTION4);
+			indicator = new Counter (prayPotIcon, plugin, "");
+		}
+	}
+	
+	// This method will go every time the config was changed and is only used for if the player deactivates the info box display so it can be properly be removed
+	public void onConfigChanged ()
+	{
+		if (!config.inInfoBox () && infoBoxManager.getInfoBoxes ().contains (indicator))
+			infoBoxManager.removeInfoBox (indicator);
+	}
+	
+	@Override
+	public Dimension render (Graphics2D graphics)
+	{
+		//Before adding it to the infoBox let's make the sure the player want's it to be there
+		if (config.inInfoBox ())
+		{
+			if (isPlayerPrayerLow ())
+			{
+				if (!infoBoxManager.getInfoBoxes ().contains (indicator))
+					infoBoxManager.addInfoBox (indicator);
+			}
+			else
+			{
+				if (infoBoxManager.getInfoBoxes ().contains (indicator))
+					infoBoxManager.removeInfoBox (indicator);
+			}
+		}
+		
+		//First let's make sure that the orb indicator actually needs be drawn
+		if (isPlayerPrayerLow ())
+		{
+			//Before drawing the orb let's make sure the player want's it to be drawn
+			if (config.displayOverOrb ())
+			{
+				//Let's get the quick xp orb widget
+				Widget quickPrayOrb = client.getWidget (WidgetInfo.QUICK_PRAYER_ORB);
+				//Make sure the prayer orb isn't deactivated
+				if (quickPrayOrb == null)
+				{
+					return null;
+				}
+				
+				//Get the bounds of the quick prayer orb
+				Rectangle2D orbBounds = quickPrayOrb.getBounds ().getBounds2D ();
```

Comment:
Why are you using getBounds2D? getBounds() seems to be enough. Then you can just use a Rectangle instead of Rectangle2D.